src/middleware.ts

```diff
@@ -10,7 +10,11 @@ async function IPXReqHandler (req: IncomingMessage, res: ServerResponse, ipx: IP
   const adapter = decodeURIComponent(urlArgs.shift() || '')
   const format = decodeURIComponent(urlArgs.shift() || '')
   const operations = decodeURIComponent(urlArgs.shift() || '')
-  const src = urlArgs.map(c => decodeURIComponent(c)).join('/')
+  let src = urlArgs.map(c => decodeURIComponent(c)).join('/')
+
+  if (adapter === 'remote' && !src.startsWith('http')) {
+    src = (req.headers.referer || req.headers.host || '') + src
```

Comment:
I think this should be implemented in image module to send full URL (PS: cache may vary based on request)

Reply:
@pi0  Moving this into the image module will force us to give the base URL.  
> cache may vary based on request  

The cacheKey generates based on src, this does not harm IPX behavior.